engine/docs/swagger/docs.go

```diff
@@ -191,6 +191,46 @@ var doc = `{
                     }
                 }
             }
+        },
+        "/api/v1/leader/resign": {
+            "post": {
+                "description": "resigns the leader",
+                "tags": [
+                    "leader"
+                ],
+                "summary": "Resign the leader",
+                "responses": {
+                    "202": {
+                        "description": ""
+                    },
+                    "400": {
+                        "description": ""
+                    },
+                    "500": {
+                        "description": ""
+                    }
+                }
+            }
+        },
+        "/api/v1/status": {
```

Comment:
To me, an interface to get who is leader and current master/executors' status is more useful.
It's a little inefficient for maintenance tools to check every server master node to get who is leader.

Reply:
Here is pd's router rules:
https://github.com/tikv/pd/blob/master/server/api/router.go#L271

Reply:
Is it enough to add an extra `leader_addr` field to the status response?

Reply:
How about using an explicit  "/api/v1/leader"  path to get the leader like PD?

I will refactor the discovery service later. At that time, I will add more membership API.

Reply:
SGTM